### PR TITLE
setting intigretion

### PR DESCRIPTION
**Pull request by vikas339**

### CODE REVIEW

File: src/components/LoginForm.vue [hunk 1]
```diff
@@ -26,7 +26,7 @@
             </div>
           </div>
           <!-- End of Form -->
-          <div class="form-group">
+           <div class="form-group">
```
Comment (chayan007) [inline src/components/LoginForm.vue:29]: Non-required space, revert back.

File: src/components/SignUp.vue [hunk 2]
```diff
@@ -69,7 +69,7 @@
           /></span>
         </button>
         <br />
-        <button class="btn btn-pill github mb-3 pad-google" type="button">
+        <button class="btn btn-pill github mb-3 pad-google g-signin2" type="button">
```
Comment (chayan007) [inline src/components/SignUp.vue:72]: is this class required ? For what purpose ?

File: src/store/modules/auth.store.js [hunk 3]
```diff
@@ -50,7 +50,13 @@ export const authStore = {
                         dispatch('alertStore/error', error, { root: true });
                     }
                 );
+        },
+        setting({ dispatch, commit }, {
+            username,
+        }) {
+            commit('registerRequest', { username });
```
Comment (chayan007) [inline src/store/modules/auth.store.js:57]: create a mutation function named `registerRequest` below.

File: src/store/modules/auth.store.js [hunk 4]
```diff
@@ -50,7 +50,13 @@ export const authStore = {
                         dispatch('alertStore/error', error, { root: true });
                     }
                 );
+        },
+        setting({ dispatch, commit }, {
+            username,
```
Comment (chayan007) [inline src/store/modules/auth.store.js:55]: pull it up in single line, refer `login` above

File: src/components/SignUp.vue (hunk 2 above)
Comment (vikas339) [reply src/components/SignUp.vue:72]: no

File: src/store/modules/auth.store.js (hunk 3 above)
Comment (vikas339) [reply src/store/modules/auth.store.js:57]: ok

File: src/components/SettingsPage.vue [hunk 5]
```diff
@@ -255,7 +255,22 @@ import AdSettings from "@/components/adset.vue";
...
   name: "SettingsPage",
   components: { AdSettings },
-};
+  data () {
+    return {
+      username: '',
+      submitted: false
+    }
+},
+methods: {
```
Comment (chayan007) [inline src/components/SettingsPage.vue:264]: `methods` is not being closed, please see to this. This is why error is being raised.

File: src/components/SettingsPage.vue [hunk 6]
```diff
@@ -26,6 +26,7 @@
                   type="file"
                   class="custom-file-input"
                   id="customFile"
+                  v-on:change="profilePicture"
```
Comment (chayan007) [inline src/components/SettingsPage.vue:29]: Rename the function to `handleProfilePicture` and also add it in methods

**Review by chayan007 — CHANGES_REQUESTED** [review]

File: src/components/SettingsPage.vue (hunk 6 above)
Comment (vikas339) [reply src/components/SettingsPage.vue:29]: change "profilePicture" to "handleProfilePicture" ?

Comment (vikas339) [reply src/components/SettingsPage.vue:29]: "profilePicture" to "handleProfilePicture"?

File: src/components/SettingsPage.vue [hunk 7]
```diff
@@ -243,18 +243,69 @@
         </div>
       </div>
       <!--End of Accordion-->
-      <button class="btn btn-primary text-success ml-3 mt-2" type="button"><a style='color:green'>Save</a></button>
-            <button class="btn btn-primary text-success ml-3 mt-2" type="button"><a style='color:red'>Discard</a></button>
-
+      <button
+        class="btn btn-primary text-success ml-3 mt-2"
+        type="button"
+        onClick="{handleSubmit}"
```
Comment (chayan007) [inline src/components/SettingsPage.vue:249]: `v-on:click="handleSubmit"`

File: src/components/SettingsPage.vue [hunk 8]
```diff
@@ -243,18 +243,69 @@
...
+      submitted: false,
+    };
+  },
+  methods: {
+    handleSubmit() {
+      debugger
+      this.submitted = true;
+      const { profilePicture, username, bio, password1, password2 } = this;
+      const { dispatch } = this.$store;
+      if (username && password1 && bio) {
```
Comment (chayan007) [inline src/components/SettingsPage.vue:283]: Why are these three fields mandatory ? Can I not just update my username ?

Comment (chayan007) [reply src/components/SettingsPage.vue:283]: Please remove entirely.

File: src/configurations/messages.js [hunk 9]
```diff
@@ -2,6 +2,7 @@ const messagingConfig = {
     messages: {
         error: {
             password_mismatch: 'Both passwords do not match. Please try again.',
+            bio_size: 'The size of your bio must be in 200 letter',
```
Comment (chayan007) [inline src/configurations/messages.js:5]: remove this and use `field_error`

File: src/store/modules/auth.store.js [hunk 10]
```diff
@@ -22,8 +22,7 @@ export const authStore = {
             userService.setting(profilePicture, username, bio, password1, password2)
             .then(
                 user => {
-                    // localStorage.setItem('user', JSON.stringify(data));
-                    localStorage.setItem('user', username);
+                    localStorage.setItem('user', JSON.stringify(data));
```
Comment (chayan007) [inline src/store/modules/auth.store.js:25]:
```
storedProfile = JSON.parse(localStorage.getItem('user'));
storedProfile.profile = user;
localStorage.setItem('user', JSON.stringify(storedProfile));
```

Comment (chayan007) [reply src/store/modules/auth.store.js:25]: You can also `console.log(user)` to check if you are getting expected response as you saw in Postman.

File: src/components/SettingsPage.vue (hunk 6 above)
Comment (chayan007) [reply src/components/SettingsPage.vue:29]: first one is a field, another is a function

File: src/services/user.service.js [hunk 11]
```diff
@@ -12,3 +15,33 @@ function getDetails() {
...
+    
+    const existingUsername = JSON.parse(localStorage.getItem('user')).profile.user.username;
+    const url =  stringFormat(`${config.commonConfig.$apiUrl}/${config.userConfig.api.settings.endpoint}`, existingUsername);
+    console.log(url);
+    console.log(registerBody);
+
+
+    return axios.put(
+        url,
+        registerBody
```
Comment (chayan007) [inline src/services/user.service.js:35]: Please use authHeader as specified in that body.

File: src/store/modules/auth.store.js [hunk 12]
```diff
@@ -10,6 +11,36 @@ export const authStore = {
     namespaced: true,
     state: initialState,
     actions: {
+        setting({ dispatch, commit }, {
```
Comment (chayan007) [inline src/store/modules/auth.store.js:14]: please move this to actions of `store/modules/user.store.js`

File: src/services/user.service.js [hunk 13]
```diff
@@ -12,3 +15,33 @@ function getDetails() {
...
+    )
+        .then((response) => {
+            const data = response.data
+            if (data.profile) {
+                localStorage.setItem('user', JSON.stringify(data));
+                return data.profile
+            }
+        })
+        .catch((error) => {
+            console.log(error);
```
Comment (chayan007) [inline src/services/user.service.js:45]: Please handle 4XX cases and 5XX cases.

File: src/services/user.service.js (hunk 11 above)
Comment (chayan007) [reply src/services/user.service.js:34]: @vikas339 please do this and commit.

File: src/services/user.service.js (hunk 13 above)
Comment (vikas339) [reply src/services/user.service.js:48]: the password and bioSize one?

File: src/services/user.service.js (hunk 11 above)
Comment (vikas339) [reply src/services/user.service.js]: done

File: src/store/modules/auth.store.js (hunk 12 above)
Comment (vikas339) [reply src/store/modules/auth.store.js:14]: done

File: src/components/SettingsPage.vue (hunk 6 above)
Comment (chayan007) [reply src/components/SettingsPage.vue:29]: @vikas339 unaddressed comment

File: src/components/SettingsPage.vue [hunk 14]
```diff
@@ -243,18 +243,82 @@
...
+        );
+      }
+      if (bio && bio.length >= 200) {
+        dispatch(
+          "alertStore/error",
+          config.messagingConfig.messages.error.bio_size,
+          { root: true }
+        );
+      }
+      dispatch("userStore/setting", {
```
Comment (chayan007) [inline src/components/SettingsPage.vue:313]: please make use of `updateFields` instead of passing an entire list of variables which are non-mandatory.

File: src/services/user.service.js [hunk 15]
```diff
@@ -12,3 +15,36 @@ function getDetails() {
     };
     return fetch(`${config.commonConfig.$apiUrl}/users`, requestOptions).then(handleResponse);
 }
+function setting(profilePicture, username, bio, password1, password2) {
+    const registerBody = {
```
Comment (chayan007) [inline src/services/user.service.js:19]: To be replaced by `updateFields`

File: src/services/user.service.js [hunk 16]
```diff
@@ -12,3 +15,36 @@ function getDetails() {
...
+        profilePicture : profilePicture,
+        username: username,
+        bio: bio,
+        password1: password1,
+        password2: password2
+    }
+    const authenticationHeader = authHeader();
+    const existingUsername = JSON.parse(localStorage.getItem('user')).profile.user.username;
+    const url =  stringFormat(`${config.commonConfig.$apiUrl}/${config.userConfig.api.settings.endpoint}`, existingUsername);
+    console.log(url);
```
Comment (chayan007) [inline src/services/user.service.js:29]: Remove `console.log()` - applicable to everywhere introduced in this PR.

File: src/services/user.service.js (hunk 13 above)
Comment (chayan007) [reply src/services/user.service.js:48]: Handle HTTP 4XX and 5XX errors when it reaches in error case.

File: src/store/modules/user.store.js [hunk 17]
```diff
@@ -14,8 +15,40 @@ export const userStore = {
...
+                    storedProfile.profile = user.updateFields;
+                    localStorage.setItem('user', JSON.stringify(storedProfile));
+                    console.log(storedProfile.updateFields);
+                    console.log(user);
+
+                    commit('loginSuccess', user);
+                    router.push('/');
+                },
+                error => {        
+                    commit('loginFailure', error);
```
Comment (chayan007) [inline src/store/modules/user.store.js:44]: Update needs no commit to `loginFailure`

File: src/store/modules/user.store.js [hunk 18]
```diff
@@ -14,8 +15,40 @@ export const userStore = {
...
+            password1,
+            password2,
+        }) {
+            commit('settingRequest', { username });
+
+            userService.setting(profilePicture, username, bio, password1, password2)
+            .then(
+                user => {
+                    // console.log(user);
+                    let storedProfile = JSON.parse(localStorage.getItem('user'));
```
Comment (chayan007) [inline src/store/modules/user.store.js:32]: This replacing (mutating) of user in localStorage needs to be done in `refreshUser` mutation.

File: src/store/modules/user.store.js [hunk 19]
```diff
@@ -14,8 +15,40 @@ export const userStore = {
...
+                    console.log(storedProfile.profile.user);
+                    storedProfile.profile = user.profile;
+                    storedProfile.profile = user.updateFields;
+                    localStorage.setItem('user', JSON.stringify(storedProfile));
+                    console.log(storedProfile.updateFields);
+                    console.log(user);
+
+                    commit('loginSuccess', user);
```
Comment (chayan007) [inline src/store/modules/user.store.js:40]: There is no requirement of this. Please remove anything you feel have blindly copied and has no use for updation.

File: src/store/modules/user.store.js [hunk 20]
```diff
@@ -14,8 +15,40 @@ export const userStore = {
...
+        setting({ dispatch, commit }, {
+            profilePicture,
+            username,
+            bio,
+            password1,
+            password2,
+        }) {
+            commit('settingRequest', { username });
+
+            userService.setting(profilePicture, username, bio, password1, password2)
```
Comment (chayan007) [inline src/store/modules/user.store.js:28]: make use of `updateFields`

File: src/store/modules/auth.store.js [hunk 21]
```diff
@@ -72,8 +74,10 @@ export const authStore = {
         logout(state) {
             state.status = {};
             state.user = null;
-        }
+        },
```
Comment (chayan007) [inline src/store/modules/auth.store.js:77]: remove trailing `,` and empty line.

File: src/store/modules/auth.store.js [hunk 22]
```diff
@@ -50,7 +51,8 @@ export const authStore = {
                         dispatch('alertStore/error', error, { root: true });
                     }
                 );
-        }
+        },
```
Comment (chayan007) [inline src/store/modules/auth.store.js:54]: remove trailing , and empty line.

File: src/store/modules/auth.store.js [hunk 23]
```diff
@@ -2,6 +2,7 @@ import { authService } from '@/services';
 import router from "@/router";
 
 const user = JSON.parse(localStorage.getItem('user'));
+// const user = null;
```
Comment (chayan007) [inline src/store/modules/auth.store.js:5]: remove this line.

File: src/components/SettingsPage.vue [hunk 24]
```diff
@@ -243,18 +243,82 @@
...
+        dispatch(
+          "alertStore/error",
+          config.messagingConfig.messages.error.password_mismatch,
+          { root: true }
+        );
+      }
+      if (bio && bio.length >= 200) {
+        dispatch(
+          "alertStore/error",
+          config.messagingConfig.messages.error.bio_size,
```
Comment (chayan007) [inline src/components/SettingsPage.vue:309]: please use error message `field_error` here.

**Comment by chayan007** [comment]
Please test this three scenario:

1. changing `username`
2. changing `password`
3. changing `bio`

There will be Token refresh in case (1) and (2) and you need to make them login again.

File: src/components/SettingsPage.vue [hunk 25]
```diff
@@ -243,18 +228,78 @@
...
 <script>
 import AdvancedSettings from "@/components/AdvancedSettings.vue";
+import { config } from "@/configurations";
+
 export default {
   name: "SettingsPage",
   components: { AdvancedSettings },
+  data() {
+    return {
+      handleProfilePicture: "",
```
Comment (chayan007) [inline src/components/SettingsPage.vue:254]: this is supposed to be a method ! You cannot alter file fields using v-model.

File: src/components/SettingsPage.vue [hunk 26]
```diff
@@ -26,6 +26,7 @@
                   type="file"
                   class="custom-file-input"
                   id="customFile"
+                  v-on:change="handleProfilePicture"
```
Comment (chayan007) [inline src/components/SettingsPage.vue:29]: use the function to add `profilePicture` 

Refer: https://stackoverflow.com/questions/41803012/v-model-doesnt-support-input-type-file

File: src/components/SettingsPage.vue [hunk 27]
```diff
@@ -243,18 +228,78 @@
...
+        updateFields["username"] = username;
+      }
+      if (password1) {
+        updateFields["password1"] = password1;
+      }
+      if (password2) {
+        updateFields["password2"] = password2;
+      }
+
+      if (password1 && password1.trim() !== password2.trim()) {
```
Comment (chayan007) [inline src/components/SettingsPage.vue:284]: Please validate password only if present. Please move it inside the `if(password1)` block.

File: src/components/SettingsPage.vue [hunk 28]
```diff
@@ -243,18 +228,78 @@
...
+        dispatch(
+          "alertStore/error",
+          config.messagingConfig.messages.error.password_mismatch,
+          { root: true }
+        );
+      }
+      if (bio && bio.length >= 200) {
```
Comment (chayan007) [inline src/components/SettingsPage.vue:291]: Please validate bio only if present. Please move it inside the if(bio) block.

File: src/components/SettingsPage.vue [hunk 29]
```diff
@@ -243,18 +228,78 @@
...
+    handleSubmit() {
+      this.submitted = true;
+      const { handleProfilePicture, username, bio, password1, password2 } = this;
+      const { dispatch } = this.$store;
+      let updateFields = {};
+      if (bio) {
+        updateFields["bio"] = bio;
+      }
+      if (handleProfilePicture) {
+        updateFields["handleProfilePicture"] = handleProfilePicture;
```
Comment (chayan007) [inline src/components/SettingsPage.vue:272]: This does not work. Please do it properly.

File: src/services/user.service.js (hunk 13 above)
Comment (chayan007) [reply src/services/user.service.js:41]: Please refer to `catch`-es in `auth.service.js`

File: src/store/modules/auth.store.js (hunk 21 above)
Comment (chayan007) [reply src/store/modules/auth.store.js:75]: Please remove empty lines @vikas339

File: src/store/modules/user.store.js (hunk 18 above)
Comment (chayan007) [reply src/store/modules/user.store.js:27]: Where is the mutation function ? @vikas339

File: src/store/modules/user.store.js [hunk 30]
```diff
@@ -14,8 +15,32 @@ export const userStore = {
                     users => commit('getAllSuccess', users),
                     error => commit('getAllFailure', error)
                 );
-        }
+        },
+        setting({ dispatch, commit }, {
+            updateFields
```
Comment (chayan007) [inline src/store/modules/user.store.js:20]: Put it in a single line.

Comment (chayan007) [reply src/store/modules/user.store.js:20]: What is the reason of keeping `updateFields` in a separate line anyway ?

**Review by chayan007 — CHANGES_REQUESTED** [review]

File: src/components/AdvancedSettings.vue [hunk 31]
```diff
@@ -26,6 +26,7 @@
                 class="form-control datepicker"
                 id="exampleInputDate1"
                 placeholder="Select date"
+                v-model="DOB"
```
Comment (chayan007) [inline src/components/AdvancedSettings.vue:29]: is this required, @vikas339  ? I thought this was merged to main Settings page.

File: src/components/SettingsPage.vue [hunk 32]
```diff
@@ -237,24 +222,478 @@
...
-  components: { AdvancedSettings },
+  components: {},
+  data() {
+    return {
+      profilePicture: "",
+      username: "",
+      bio: "",
+      password1: "",
+      password2: "",
+      DOB: "",
```
Comment (chayan007) [inline src/components/SettingsPage.vue:602]: variables should be in camelCase ! Please rename.

File: src/components/SettingsPage.vue [hunk 33]
```diff
@@ -237,24 +222,478 @@
...
+      contact: "",
+      address: "",
+      profession: "",
+      gender_male: "",
```
Comment (chayan007) [inline src/components/SettingsPage.vue:606]: it should be one variable:

```
gender
relationship_status
```

File: src/components/SettingsPage.vue [hunk 34]
```diff
@@ -237,24 +222,478 @@
...
+                            name="exampleRadios1"
+                            id="exampleRadios7"
+                            value="divorced"
+                          />
+                          <label class="form-check-label" for="exampleRadios7">
+                            Divorced
+                          </label>
+                        </div>
+                        <div class="form-check">
+                          <input
```
Comment (chayan007) [inline src/components/SettingsPage.vue:542]: @itisdb / @ayushi-siddhu why are choice field form elements treated as inputs and not as options ? Please don't use any random element to fit into template which turns out to be actual useless at future while integration. Extra time is involved now to repair it and it becomes dependency to some other developers as well !

File: src/services/user.service.js [hunk 35]
```diff
@@ -12,3 +15,29 @@ function getDetails() {
...
+    const existingUsername = JSON.parse(localStorage.getItem('user')).profile.user.username;
+    const url =  stringFormat(`${config.commonConfig.$apiUrl}/${config.userConfig.api.settings.endpoint}`, existingUsername);
+
+    return axios.put(
+        url,
+        registerBody,
+        { headers: authenticationHeader } 
+    )
+        .then(response => {
+            console.log(response)
```
Comment (chayan007) [inline src/services/user.service.js:32]: you can remove the console.log

File: src/services/user.service.js [hunk 36]
```diff
@@ -12,3 +15,29 @@ function getDetails() {
     };
     return fetch(`${config.commonConfig.$apiUrl}/users`, requestOptions).then(handleResponse);
 }
+function setting(updateFields) {
```
Comment (chayan007) [inline src/services/user.service.js:18]: *settings

File: src/store/modules/user.store.js (hunk 18 above)
Comment (chayan007) [reply src/store/modules/user.store.js:27]: Why is this still not moved into `refreshUser` mutation ?

File: src/services/user.service.js [hunk 37]
```diff
@@ -12,3 +15,29 @@ function getDetails() {
...
+    return axios.put(
+        url,
+        registerBody,
+        { headers: authenticationHeader } 
+    )
+        .then(response => {
+            console.log(response)
+            const data = response.data
+            if (data.profile) {
+                localStorage.setItem('user', JSON.stringify(data));
```
Comment (chayan007) [inline src/services/user.service.js:35]: Why are you re-storing the data and removing the token ? @vikas339

File: src/store/modules/auth.store.js [hunk 38]
```diff
@@ -1,7 +1,7 @@
 import { authService } from '@/services';
 import router from "@/router";
 
-const user = JSON.parse(localStorage.getItem('user'));
+let user = JSON.parse(localStorage.getItem('user'));
```
Comment (chayan007) [inline src/store/modules/auth.store.js:4]: Why is this line required in this file ?

File: src/store/modules/user.store.js (hunk 18 above)
Comment (chayan007) [reply src/store/modules/user.store.js:27]: If already moved, what is this still doing here ?

File: src/store/modules/user.store.js [hunk 39]
```diff
@@ -25,6 +50,27 @@ export const userStore = {
...
+            state.user = user;
+        },
+        loginFailure(state) {
+            state.status = {};
+            state.user = null;
+        },
+        refreshUser(state, error){
+            state.all = { error};
+            let user=localStorage.getItem("user");
+            user.profile.user;
```
Comment (chayan007) [inline src/store/modules/user.store.js:69]: What is the meaning of this line `user.profile.user;` ?

File: src/store/modules/user.store.js [hunk 40]
```diff
@@ -25,6 +50,27 @@ export const userStore = {
...
+            state.status = { loggedIn: true };
+            state.user = user;
+        },
+        loginFailure(state) {
+            state.status = {};
+            state.user = null;
+        },
+        refreshUser(state, error){
+            state.all = { error};
+            let user=localStorage.getItem("user");
```
Comment (chayan007) [inline src/store/modules/user.store.js:68]: in one line you are using `"` and on the next line you are using `'` ? Please don't code like a school kid anymore.

File: src/components/SettingsPage.vue (hunk 29 above)
Comment (chayan007) [reply src/components/SettingsPage.vue:640]: Still this does not work, please do it properly.

Comment (chayan007) [reply src/components/SettingsPage.vue:640]: the name of the field is `profilePicture`

Comment (chayan007) [reply src/components/SettingsPage.vue:640]: Unaddressed

**Comment by chayan007** [comment]
@vikas339 please run the following test:

1. Change profile picture
2. Change gender
3. Change bio and relationship status
4. Change profession

Please let me know all the test passes.

File: src/components/SettingsPage.vue [hunk 41]
```diff
@@ -237,24 +222,479 @@
...
+        updateFields["bio"] = bio;
+        if (bio.length >= 200) {
+          dispatch(
+            "alertStore/error",
+            config.messagingConfig.messages.error.field_error,
+            { root: true }
+          );
+        }
+      }
+      if (handleProfilePicture) {
```
Comment (chayan007) [inline src/components/SettingsPage.vue:660]: `if profilePicture` ?

**Review by chayan007 — APPROVED** [review]

**Review by itisdb — APPROVED** [review]
